internal/store/utils.go

```diff
@@ -30,6 +30,9 @@ import (
 	"k8s.io/kube-state-metrics/v2/pkg/options"
 )
 
+// SharedLabelKeys is the type for labelKeys which are shared in all metrics under one resource type
+type SharedLabelKeys []string
```

Comment:
What is the benefit of making this a type?

Reply:
It's easier to parse using static analysis?

Reply:
Yes, it makes parsing easier.

Find variable with type SharedLabelKeys, then append these shared labels into each ingress metric.

https://github.com/kubernetes/kube-state-metrics/blob/c8c94159b10ca8c4ab55a0fb2dd0cfd96fd15ef6/internal/store/ingress.go#L41